src/lib.rs

```diff
@@ -1160,6 +1164,9 @@ where
                         } else {
                             return None;
                         };
+
+                        let ip : &str = if ip != "" {&ip} else {&host.as_ref().unwrap()};
```

Comment:
We don't need type hint here? 
```suggestion
                        let ip = if ip != "" {&ip} else {&host.as_ref().unwrap()};
```